modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/UserResourceManager.java

```diff
@@ -358,6 +363,100 @@ public SCIMResponse listWithGET(UserManager userManager, String filter,
         }
     }
 
+    /**
+     * Build Node for filtering.
+     *
+     * @param filter Filter in the request.
+     * @param schema Schema
+     * @return Node
+     * @throws BadRequestException
+     * @throws IOException
+     */
+    private Node buildNode(String filter, SCIMResourceTypeSchema schema) throws BadRequestException, IOException {
+
+        if (filter != null) {
+            FilterTreeManager filterTreeManager = new FilterTreeManager(filter, schema);
+            return filterTreeManager.buildTree();
+        }
+        return null;
+    }
+
+    /**
+     * Resolves the sorting order of the filter.
+     *
+     * @param sortOrder Sort order in the request.
+     * @param sortBy    SortBy in the request.
+     * @return Resolved sorting order.
+     * @throws BadRequestException Invalid sorting order.
+     */
+    private String resolveSortOrder(String sortOrder, String sortBy) throws BadRequestException {
+
+        if (sortOrder != null) {
+            if (!(sortOrder.equalsIgnoreCase(SCIMConstants.OperationalConstants.ASCENDING) || sortOrder
+                    .equalsIgnoreCase(SCIMConstants.OperationalConstants.DESCENDING))) {
+                String error = " Invalid sortOrder value is specified";
+                throw new BadRequestException(error, ResponseCodeConstants.INVALID_VALUE);
+            }
+        }
+        // If a value for "sortBy" is provided and no "sortOrder" is specified, "sortOrder" SHALL default to
+        // ascending.
+        if (sortOrder == null && sortBy != null) {
```

Comment:
use stringutils

Reply:
apache-commons in not in Charon. Therefore, unable to user StringUtils.